kernelci/build.py

```diff
@@ -450,15 +450,20 @@ def _run_make(kdir, arch, target=None, jopt=None, silent=True, cc='gcc',
     return shell_cmd(cmd, True)
 
 
-def _make_defconfig(defconfig, kwargs, fragments, verbose, log_file):
+def _make_defconfig(defconfig, kwargs, fragments, verbose, log_file, opts):
     kdir, output_path = (kwargs.get(k) for k in ('kdir', 'output'))
     result = True
 
+    defconfig_kwargs = dict(kwargs)
+    defconfig_opts = dict(defconfig_kwargs['opts'])
+    defconfig_kwargs['opts'] = defconfig_opts
     tmpfile_fd, tmpfile_path = tempfile.mkstemp(prefix='kconfig-')
     tmpfile = os.fdopen(tmpfile_fd, 'w')
     defs = defconfig.split('+')
     target = defs.pop(0)
     for d in defs:
+        if d.startswith('KCONFIG_ALLCONFIG'):
```

Comment:
Looks good for this use-case.  If we see some value in passing other `KCONFIG_*` things in the full defconfig name then we may want to do something a bit more general such as:
```python
if d.startswith('KCONFIG_'):
    config, value = d.split('=')
    defconfig_opts[config] = value
```